### PR TITLE
[jenkins] feature: run CI pipeline in docker

### PR DESCRIPTION
**Pull request by Jaguar0625**
provides better isolation

### CODE REVIEW

File: jenkins/docker/javascript.Dockerfile [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+RUN curl -fsSL https://deb.nodesource.com/setup_16.x | bash - \
+	&& apt-get install -y nodejs
+
+# rest dependencies
+RUN apt-get install -y make gcc g++
+
+# add ubuntu user (used by jenkins)
+RUN useradd -ms /bin/bash ubuntu
+
+USER ubuntu
```
Comment (gimre-xymcity) [inline jenkins/docker/javascript.Dockerfile:18]: we'll discuss later, but neither of those two should be needed
(i.e. USER always forces given uid, and we probably want to use `--user` from outside like we did)

**Review by gimre-xymcity — CHANGES_REQUESTED** [review]
rest of the change looks ok

File: client/rest/Jenkinsfile [hunk 2]
```diff
@@ -1,7 +1,9 @@
 defaultCiPipeline {
 	platform = ['ubuntu']
-	gitHubId = 'Symbol-Github-app'
+	ciBuildDockerfile = 'javascript.Dockerfile'
+
 	packageId = 'client-rest'
+
 	publisher = 'docker'
-	docker_image_name = 'symbolplatform/symbol-rest'
+	// dockerImageName = 'symbolplatform/symbol-rest'
```
Comment (Wayonb) [inline client/rest/Jenkinsfile:8]: What's the plan for publishing images?

**Review by Wayonb — APPROVED** [review]

File: client/rest/Jenkinsfile (hunk 2 above)
Comment (Jaguar0625) [reply client/rest/Jenkinsfile:8]: idk, i just noticed that wasn't being used (and it was snake_case), so commented it out for now

File: jenkins/shared-library/vars/defaultCiPipeline.groovy [hunk 3]
```diff
@@ -63,6 +67,7 @@ void call(Closure body) {
 						steps {
 							println("Parameters: ${params}")
 							sh 'printenv'
+							sh 'ls -alh /home/ubuntu'
```
Comment (gimre-xymcity) [inline jenkins/shared-library/vars/defaultCiPipeline.groovy:70]: drop ^^

**Review by gimre-xymcity — APPROVED** [review]

**Pull request merged**